src/app/(auth)/subscriber/edit/page.tsx

```diff
@@ -0,0 +1,67 @@
+'use client';
+
+import Subscriber from '@/components/Subscriber/Subscriber';
+import { UserGroupIcon } from '@heroicons/react/24/outline';
+import { updateSubscriber, getSubscriber } from '@/apis/subscriber';
+import { useEffect, useState } from 'react';
+import { toast } from 'react-toastify';
+import { useRouter, useSearchParams } from 'next/navigation';
+
+const Page = () => {
+  const searchParams = useSearchParams();
+  const id = searchParams.get('id');
+  const router = useRouter();
+  const [selectedOptions, setSelectedOptions] = useState([]);
+  const [subscriber, setSubscriber] = useState({});
+
+  useEffect(() => {
+    const fetchSubscriber = async () => {
+      try {
+        const { data } = await getSubscriber(id);
+        setSelectedOptions(
+          data.tags.map((tag: { name: string; id: number }) => {
+            return { value: tag.id, label: tag.name };
+          })
+        );
+        setSubscriber(data);
+      } catch (error: any) {
+        toast.error(error.response.data.message);
+      }
+    };
+    fetchSubscriber();
+  }, []);
+
+  const onSubmit = async (values: any) => {
+    if (selectedOptions.length > 0) {
+      values.tags = selectedOptions.map(
+        (selectedOption: { value: any }) => selectedOption.value
+      );
+    }
```

Comment:
this code looks repetitive in both create and update page. I suggest to keep it inside the common form component and pass the required values to the parent's `onSubmit` function

Reply:
Keeping in mind the SRP and DRY principles, I have made some changes. You check once. And let me know, it's fine and still we need to some more changes.